README.md

```diff
@@ -8,36 +8,47 @@ A service mesh for polyglot microservices.
 
 ## Introduction
 
-Istio is an open source system providing a uniform way to manage and connect
-microservices. It is composed of:
-*  A proxy handling service-to-service and external-to-service traffic.
-*  A mixer supporting access checks, quota allocation and deallocation,
-monitoring and logging.
-*  A manager handling system configuration, discovery, and automation.
-
-The [architectural overview](ARCHITECTURE.md) provides a high-level summary of
-the design. The [milestone plan](MILESTONES.md) gives a rough estimate of 
-what we expect to release and when.
+Istio is an open platform for providing a uniform way to integrate
+microservices, manage traffic flow across microservices, enforce policies
+and aggregate telemetry data. Istio's control plane provides an abstraction
+layer over the underlying cluster management platform, such as Kubernetes,
+Mesos, etc.
+
+Istio is composed of three main components:
+
+* **Proxy** - Sidecars per microservice to handle ingress/egress traffic
+   between services in the cluster and from a service to external
+   services. The proxies form a _secure Layer-7 microservice mesh_ enabling a rich variety of
+   traffic management, policy enforcement and telemetry recording/reporting
+   functions.
+* **Mixer** - Central component that co-ordinates with various proxies to
+   enforce policies such as ACLs, rate limits, authentication, request tracing
+   and metrics collection.
+* **Manager** - A configuration manager responsible for configuring the
+  proxies and the mixer at runtime.
+
+A high-level overview of various components in Istio is available [here](doc/overview.md).
 
 ## Repositories
 
 The Istio project is divided across multiple GitHub repositories. Each
 repository contains information about how to build and test it.
 
 - [istio/api](https://github.com/istio/api). This repository defines
-component-level APIs and common configuration  formats for the Istio platform.
+component-level APIs and common configuration formats for the Istio platform.
 
-- [istio/istio](https://github.com/istio/istio). The main Istio repo which is
-used to host the high-level documentation for the project, along with 
+- [istio/istio](https://github.com/istio/istio). The main Istio repo which
```

Comment:
This is sort of a self link. Maybe should mention that Its README is this document.